tools/gen-strings.py

```diff
@@ -57,38 +62,45 @@ def read_magic_string_defs(debug=False):
     # sorted by length and alpha.
     ini_parser = ConfigParser()
     ini_parser.optionxform = str # case sensitive options (magic string IDs)
-    ini_parser.read(MAGIC_STRINGS_INI)
+    ini_parser.read(ini_path)
 
     defs = [(str_ref, json.loads(str_value) if str_value != '' else '')
-            for str_ref, str_value in ini_parser.items('LIT_MAGIC_STRINGS')]
+            for str_ref, str_value in ini_parser.items(item_name)]
     defs = sorted(defs, key=lambda ref_value: (len(ref_value[1]), ref_value[1]))
 
+    if len(defs[-1][1]) > LIMIT_MAGIC_STR_LENGTH:
+        for str_ref, str_value in [x for x in defs if len(x[1]) > LIMIT_MAGIC_STR_LENGTH]:
+            print("error: The maximum allowed magic string size is {limit} but {str_ref} is {str_len} long.".format(
+                limit=LIMIT_MAGIC_STR_LENGTH, str_ref=str_ref, str_len=len(str_value)))
+        exit(1)
+
     if debug:
         print('debug: magic string definitions: {dump}'
               .format(dump=debug_dump(defs)))
 
     return defs
 
 
-def extract_magic_string_refs(debug=False):
+def extract_magic_string_refs(debug, pattern, inc_h_filename):
     results = {}
 
-    def process_line(fname, lnum, line, guard_stack):
+    def process_line(fname, lnum, line, guard_stack, pattern):
         # Build `results` dictionary as
         #   results['LIT_MAGIC_STRING_xxx'][('!defined (CONFIG_DISABLE_yyy_BUILTIN)', ...)]
         #       = [('zzz.c', 123), ...]
         # meaning that the given literal is referenced under the given guards at
         # the listed (file, line number) locations.
-        for str_ref in re.findall('LIT_MAGIC_STRING_[a-zA-Z0-9_]+', line):
-            if str_ref in ['LIT_MAGIC_STRING_DEF',
-                           'LIT_MAGIC_STRING_FIRST_STRING_WITH_SIZE',
-                           'LIT_MAGIC_STRING_LENGTH_LIMIT',
-                           'LIT_MAGIC_STRING__COUNT']:
+        for str_ref in re.findall('%s_[a-zA-Z0-9_]+' % pattern, line):
+            if str_ref in ['%s_DEF' % pattern,
+                           '%s_FIRST_STRING_WITH_SIZE' % pattern,
+                           '%s_LENGTH_LIMIT' % pattern,
+                           '%s__COUNT' % pattern]:
```

Comment:
This list should be constructed before the loop and there is should be just used.